### PR TITLE
CAT-1301_permissions

### PR DESCRIPTION
**Pull request by X4mp**
added Project.getRequiredResources for reading the resources required by the current Project which is saved in saveProject and write those permissions as String to an additional file.

### CODE REVIEW

**Comment by X4mp** [comment]
linked to #1177

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

File: catroid/src/org/catrobat/catroid/io/StorageHandler.java [hunk 1]
```diff
@@ -393,6 +397,16 @@ public boolean saveProject(Project project) {
 			writer = new BufferedWriter(new FileWriter(tmpCodeFile), Constants.BUFFER_8K);
 			writer.write(projectXml);
 			writer.flush();
+
+			File permissionFile = new File(buildProjectPath(project.getName()), PROJECTPERMISSIONS_NAME);
+			writer = new BufferedWriter(new FileWriter(permissionFile), Constants.BUFFER_8K);
+
+			for ( String resource : generatePermissionsSetFromResource(project.getRequiredResources())) {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/io/StorageHandler.java]: Remove unnecessary space before String

File: catroidTest/src/org/catrobat/catroid/test/io/StorageHandlerTest.java [hunk 2]
```diff
@@ -285,6 +303,37 @@ private Float interpretFormula(Formula formula, Sprite sprite) {
         return Float.NaN;
     }
 
+	public void testGetRequiredResources() throws IOException {
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/test/io/StorageHandlerTest.java]: I guess this doesn't throw an IOException...
This test should be in the ProjectTest. But other project tests aren't in the ProjectTest either... So lets stick to the "rules" and keep it here.

File: catroidTest/src/org/catrobat/catroid/test/io/StorageHandlerTest.java [hunk 3]
```diff
@@ -285,6 +303,37 @@ private Float interpretFormula(Formula formula, Sprite sprite) {
...
+		int resources = generateMultiplePermissionsProject().getRequiredResources();
+		assertEquals("Sum over required resources not matching", Brick.ARDRONE_SUPPORT
+				| Brick.FACE_DETECTION
+				| Brick.BLUETOOTH_LEGO_NXT
+				| Brick.TEXT_TO_SPEECH, resources);
+
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/test/io/StorageHandlerTest.java]: Remove unnecessary newline

**Comment by robertpainsi** [comment]
Please do me a favor, run the auto formatter over your changes (not the whole files!). Ctrl+Alt+L and choose 'Selected text'

Also restyle and especially reword your commit message completely. http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Comment by X4mp** [comment]
Dude, trolling is not accepted on Github. I won't run Ctrl+Alt+L since this is Screen-Lock and it won't change any code. Please think b4 trolling

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Pull request merged**